src/main/java/com/griddynamics/cloud/learning/service/ArticleService.java

```diff
@@ -2,15 +2,120 @@
 
 import com.griddynamics.cloud.learning.dao.domain.Article;
 import com.griddynamics.cloud.learning.dao.domain.PaginatedResult;
+import com.griddynamics.cloud.learning.dao.repository.ArticleRepository;
+import com.griddynamics.cloud.learning.web.dto.ArticleDto;
 import com.griddynamics.cloud.learning.web.dto.ArticleFilterRequest;
+import com.griddynamics.cloud.learning.web.dto.PaginatedRequestParams;
+import com.griddynamics.cloud.learning.web.dto.PurchaseType;
 
-import java.util.Collections;
+import java.util.stream.Collectors;
 
 public class ArticleService {
 
-    //TODO (Just a stub for now; part of service implementation task)
-    public PaginatedResult<Article> getArticles(ArticleFilterRequest requestFilter) {
+    private final ArticleRepository repository;
 
-        return new PaginatedResult<>(Collections.emptyList(), 0);
+    public ArticleService() {
+        this(new ArticleRepository());
+    }
+
+    public ArticleService(ArticleRepository repository) {
+        this.repository = repository;
+    }
+
+    /**
+     * Gets articles by given filter and sorting params.
+     *
+     * @param requestFilter - request dto, that contains sorting and filtering info.
+     * @return - paginated list of articles.
+     */
+
+    public PaginatedResult<ArticleDto> findArticles(ArticleFilterRequest requestFilter) {
+
+        PaginatedResult<ArticleDto> articles;
+
+        final PurchaseType purchaseType = requestFilter.getPurchaseType();
+        final String tag = requestFilter.getTag();
+        final String caption = requestFilter.getCaption();
+        final PaginatedRequestParams params = requestFilter.getParams();
+
+        if (purchaseType != PurchaseType.ALL) {
+            articles = getArticlesByPurchaseType(tag, caption, params, purchaseType == PurchaseType.FREE);
+        } else {
+            articles = getArticles(tag, caption, params);
+        }
+        return articles;
+    }
+
+    /**
+     * Returns {@code PaginatedResult<Article>} dto from repository.
```

Comment:
-> Returns paginated articles by filters

Jvav doc description should be more meaningful. 
Same for the other method